### PR TITLE
[Feature] add metric of the number of CN

### PR DESCRIPTION
**Pull request by goldenbean**
add metric of the number of CN

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java [hunk 1]
```diff
@@ -787,6 +787,25 @@ public List<Long> getComputeNodeIds(boolean needAlive) {
...
+        return getDecommissionedComputeNodeIds().size();
+    }
+
+    public List<Long> getDecommissionedComputeNodeIds() {
+        List<Long> computeNodeIds = Lists.newArrayList(idToComputeNodeRef.keySet());
+
+        Iterator<Long> iter = computeNodeIds.iterator();
+        while (iter.hasNext()) {
+            ComputeNode computeNode = this.getComputeNode(iter.next());
+            if (computeNode == null || !computeNode.isDecommissioned()) {
```
Comment (kevincai) [inline fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java:800]: I suppose this is just for interface compatibility, right now `decommission` statement doesn't support decommissioning a compute node, is it correct?

Comment (goldenbean) [reply fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java:800]: it supposed to just drop cn, and no longer need operate decommission. I would like to remove this. :)

Comment (kevincai) [reply fe/fe-core/src/main/java/com/starrocks/system/SystemInfoService.java]: ok. actually we are planning to support cn decommission as well in near feature. lol.

it is ok for now.

**Review by kevincai — APPROVED** [review]

**Comment by abc982627271** [comment]
ut failed

**Comment by goldenbean** [comment]
> ut failed

It's weird. In fact, the changes I made has no relation with the failed UT. Could you please verify this issue ?

**Comment by goldenbean** [comment]
@abc982627271 UT passed! :)

**Review by kevincai — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/metric/PrometheusMetricVisitor.java [hunk 2]
```diff
@@ -219,6 +219,13 @@ public void getNodeInfo() {
                                 .count())
                 .append("\n");
 
+        sb.append(NODE_INFO).append("{type=\"cn_node_num\", state=\"total\"} ")
+            .append(GlobalStateMgr.getCurrentSystemInfo().getTotalComputeNodeNumber()).append("\n");
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/metric/PrometheusMetricVisitor.java:223]: need rebase

**Review by nshangyiming — APPROVED** [review]

**Comment by kevincai** [comment]
@mergifyio backport branch-3.2 branch-3.1

**Pull request merged**